PowerShell/CipherTrustManager/Connections/CipherTrustManager-Connections-IdP.psm1

```
@@ -0,0 +1,1465 @@
+    .LINK
+        https://github.com/thalescpl-io/CDSP_Orchestration/tree/main/PowerShell/CipherTrustManager
+#>
+function New-CMIdPConnectionLDAP{
+    param(
+        [Parameter(Mandatory = $true,
+        ValueFromPipelineByPropertyName = $true)]
+        [string] $name,
+        [Parameter(Mandatory)] [string] $root_dn,
+        [Parameter(Mandatory)] [string] $server_url,
+        [Parameter()] [string] $uid_field="sAMAccountName",
+        [Parameter()] [switch] $disable_auto_create,
+        [Parameter()] [string] $bind_dn, 
+        [Parameter()] [string] $bind_pass, 
+        [Parameter()] [pscredential] $bindsecurecredentials,
+        [Parameter()] [string] $group_base_dn,
+        [Parameter()] [string] $group_filter,
+        [Parameter()] [string] $group_id_field,
+        [Parameter()] [string] $group_member_field,
+        [Parameter()] [string] $guid_field,
+        [Parameter()] [string] $search_filter,
+        [Parameter()] [string] $user_dn_field,
+        [Parameter()] [string[]] $root_cas,
+        [Parameter()] [string] $root_ca_file,
+        [Parameter()] [switch] $insecure_skip_verify
+    )
+
+    Write-Debug "Start: $($MyInvocation.MyCommand.Name)"
+
+    Write-Debug "Creating an LDAP IdP Connection in CM"
+    $endpoint = $CM_Session.REST_URL + $target_uri
+    Write-Debug "Endpoint: $($endpoint)"
+
+    # Mandatory Parameters
+    $body = [ordered] @{
+        "name"          = $name
+        "strategy"      = "ldap"
+        "ldap_options"  = @{}
+    }
+
+    $body.ldap_options.add("root_dn",$root_dn)
+    $body.ldap_options.add("server_url",$server_url)
+    $body.ldap_options.add("uid_field",$uid_field)
+
+    if ($server_url.Substring(4,1) -eq "s"){
+        if(!$root_cas -and !$root_ca_file){ 
+            return "Missing LDAPS Certificate. Please try again."
+        }
+        if($root_ca_file){
+            $root_cas = Get-Content -Path $root_ca_file -raw -ErrorAction Stop
```

Thinking aloud here... can we do or do we need to validate the CA cert here to fail fast in case of a bad cert? Before hitting the CM API... I do not even know if we need to do this...just a thought